[EDU-1631] Spaces Component Locking example (#2290)
ably/docs: examples/spaces-component-locking/page.md

# Locking components within collaborative applications

> Users can lock components within a collaborative application.

Reviewer: ```suggestion
Enable users to lock components for editing within a collaborative application.
```

> Manage concurrent access to shared resources within your collaborative application. This feature is particularly useful for preventing conflicts in collaborative environments, ensuring that only one user can modify a specific component at a time.

Reviewer: ```suggestion
Manage concurrent access to shared resources within your application and prevent conflicts in collaborative environments, ensuring that only one user can modify a specific component at a time.
```

> Component locking enhances various collaborative applications, such as, in document editing tools, it can prevent simultaneous edits to the same section. In project management software, it can ensure that task details are updated by one team member at a time, maintaining data integrity and preventing conflicting changes.

Reviewer: ```suggestion
Component locking enhances applications where multiple users can edit the same fields, such as spreadsheets, slide shows and forms. It is a key component in maintaining data integrity and preventing conflicting changes.
```

> Component locking are implemented using [Ably Spaces](https://ably.com/docs/products/spaces). The Spaces SDK contains a set of purpose-built APIs that abstract away the complexities involved in managing the state of components accessibility in a collaborative application. It is built on top of Ably's core platform, and so it provides the same performance and scaling guarantees.

Reviewer: ```suggestion
Component locking is implemented using [Ably Spaces](https://ably.com/docs/products/spaces). The Spaces SDK contains a set of purpose-built APIs that abstract away the complexities involved in managing the state of components accessibility in a collaborative application. It is built on top of Ably's core platform, and so it provides the same performance and scaling guarantees.
```

## Resources

// React brief (Only visible if viewing the React example)

Use the following components to add component locking into an application:

* [`SpacesProvider`](https://ably.com/docs/spaces/react#spaces-provider): initializes and manages a shared space client instance, passing it down through React context to enable real-time spaces functionality across the application.

> * [`SpaceProvider`](https://ably.com/docs/spaces/react#spaces-provider): manages the state and functionality of a specific space, providing access to real-time interactions within that space via React context.

Reviewer: real-time --> realtime
